File: discord-collector/src/signal-pipeline-log.js
/**
 * Telegram 信号全链路日志（collect:ui 侧：归档 → 推 TG → CDP）。
 */
export const PIPELINE_TAG = "[signal-pipeline]";
export const SEND_CDP_TAG = "[send-cdp]";

/**
 * @param {string} stage
 * @param {Record<string, unknown>} [fields]
 */
export function formatPipelineLog(stage, fields = {}) {
  const parts = [PIPELINE_TAG, `stage=${stage}`];
  for (const key of [
    "cardId",
    "channelId",
    "channelName",
    "symbol",
    "direction",
    "sourceRef",
    "messageRef",
    "event",
    "reason",
    "detail",
    "jobId",
    "chatId",
  ]) {
    const val = fields[key];
    if (val == null || val === "") continue;
    parts.push(`${key}=${val}`);
  }
  return parts.join(" ");
}

/**
 * @param {string} stage
 * @param {Record<string, unknown>} [fields]
 */
export function formatSendCdpLog(stage, fields = {}) {
  return `${SEND_CDP_TAG} ${formatPipelineLog(stage, fields).slice(PIPELINE_TAG.length + 1)}`;
}

File: telegram/signal_pipeline_log.py
"""Telegram 信号全链路日志：从收消息 → 解析 → 建卡 API。"""

from __future__ import annotations

TAG = "[signal-pipeline]"


def _preview(text: str, limit: int = 120) -> str:
    s = (text or "").replace("\n", " ").strip()
    if len(s) > limit:
        return s[:limit] + "…"
    return s


def log_pipeline(stage: str, **fields: object) -> None:
    """stage: received | parsed | skip | card_post | card_ok | card_fail | dedup | pending"""
    parts: list[str] = [TAG, f"stage={stage}"]
    for key in (
        "chat_id",
        "msg_id",
        "sender",
        "symbol",
        "direction",
        "phase",
        "card_id",
        "source_ref",
        "reason",
        "detail",
        "url",
    ):
        val = fields.get(key)
        if val is None or val == "":
            continue
        parts.append(f"{key}={val!r}" if key in ("sender", "reason", "detail") else f"{key}={val}")
    body = fields.get("body")
    if body:
        parts.append(f"body={_preview(str(body))!r}")
    print(" ".join(parts), flush=True)
